.github/workflows/integration_tests.yml

```diff
@@ -169,7 +171,13 @@ jobs:
 
         # To feed input into the job matrix, we first need to convert to a JSON
         # list. Then we can use fromJson to define the field in the matrix for the tests job.
-        apis=$( python scripts/gha/print_matrix_configuration.py -c -w integration_tests -k apis -o "${{github.event.inputs.apis}}" ${TEST_MATRIX_PARAM} )
+        if [[ "${{ github.event.schedule }}" == "0 9 * * *" ]]; then
```

Comment:
I HATE doing it this way. Hate it hate it hate it.

That said, can you think of any other way to do this? :-/

Reply:
For `scheduled` jobs, There is no Webhook payload can be attached to Github Context. This is why I cannot use the {{ github.events.inputs.(field) }}. And this code is based on the example below: 
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

However, I'm aware of this as it cause confusion. And there are few things I could do to improve:
1) add more comments.
2) put this part of logic into `print_matrix_configuration.py`.
3) Having a separate cron job, which leverages `it_workflow.py` to trigger the nightly integration tests.

Please let me know what do you think.

Reply:
I think for now let's add more comments, since refactoring our workflows much more than that seems out of the scope.

Reply:
Comments added. Should I have a ticket to refactor this part later?